core/src/main/java/org/infinispan/util/concurrent/CompletableFutures.java

```diff
@@ -36,6 +40,23 @@
       sink.setFuture(compoundSource);
    }
 
+   public static <T> CompletableFuture<T> connect(NotifyingFuture<T> source) {
+      final CompletableFuture<T> result = new CompletableFuture<>();
+      source.attachListener(f -> {
+         try {
+            result.complete(f.get());
+         } catch (Exception e) {
+            result.cancel(false);
+         }
+      });
+      return result;
+   }
+
+   public static <T> CompletableFuture<List<T>> combine(List<CompletableFuture<T>> futures) {
```

Comment:
In the functional programming world (e.g. [Haskell]([here](https://hackage.haskell.org/package/base-4.8.1.0/docs/Control-Monad.html#v:sequence))), this operation is more commonly known as `sequence` which is defined as:

    sequence :: (Traversable t, Monad m) => t (m a) -> m (t a)

^ Above is Haskell code, but substitute `Traversable` for `List`, and `Monad` for `CompletableFuture`, and what that function essentially says is that it transforms a List of CompletableFuture instances into a CompletableFuture that contains a List.

Hence, I'd suggest that you rename it to `sequence`.